app/models.py

```diff
@@ -3152,4 +3156,41 @@ class PartnerUser(Base, ModelMixin):
     )
 
 
+class PartnerSubscription(Base, ModelMixin):
+    """
+    For users who have a subscription via a partner
+    """
+
+    __tablename__ = "partner_subscription"
+
+    partner_user_id = sa.Column(
+        sa.ForeignKey(PartnerUser.id, ondelete="cascade"), nullable=False, unique=True
+    )
+
+    # when the partner subscription ends
+    end_at = sa.Column(ArrowType, nullable=False)
+
+    partner_user = orm.relationship(PartnerUser)
+
+    @classmethod
+    def find_by_user_id(cls, user_id: int) -> Optional[PartnerSubscription]:
+        res = (
+            Session.query(PartnerSubscription, PartnerUser)
+            .filter(
+                and_(
+                    PartnerUser.user_id == user_id,
+                    PartnerSubscription.partner_user_id == PartnerUser.id,
+                )
+            )
+            .first()
+        )
+        if res:
+            subscription, partner_user = res
+            return subscription
+        return None
+
+    def is_active(self):
+        return self.end_at > arrow.now()
```

Comment:
should we take into account the grace period here? For example this is used for AppleSubscription:
https://github.com/simple-login/app/blob/dba56f0dae89861b64b9873626828b9ba4094bc3/app/models.py#L1976

Reply:
Added a grace period

Reply:
Good catch :ok_hand: